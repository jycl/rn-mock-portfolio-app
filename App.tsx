/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * Generated with the TypeScript template
 * https://github.com/react-native-community/react-native-template-typescript
 *
 * @format
 */
import 'react-native-gesture-handler'; // react-navigation setup
import React, { ReactNode } from 'react';
import { Provider } from 'react-redux';
import { View, Text, StyleSheet, StatusBar } from 'react-native';
import { NavigationContainer } from '@react-navigation/native';
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
import { SafeAreaProvider, SafeAreaView } from 'react-native-safe-area-context';
import PortfolioScreen from './src/screens/PortfolioScreen';
import Icon from './src/components/Icon';
import configureStore from './src/store';

const store = configureStore();

export type TabBarIconProps = {
  color: string;
  size: number;
  focused: boolean;
};

const Tab = createBottomTabNavigator();
export const GenericScreen = (props: any) => {
  const screen = props?.route?.name || 'Generic';
  return (
    <View style={styles.screen}>
      <Text style={styles.text}>{screen}</Text>
    </View>
  );
};

const App = (): ReactNode => {
  return (
    // outer SafeAreaView needed for bottom for notch iOS devices
    <Provider store={store}>
      <SafeAreaView style={styles.appContainer}>
        <SafeAreaProvider>
          <StatusBar barStyle="light-content" />
          <NavigationContainer>
            <Tab.Navigator
              sceneContainerStyle={styles.scene}
              tabBarOptions={{
                tabStyle: { backgroundColor: 'black' },
                activeTintColor: 'white',
              }}>
              <Tab.Screen
                name="Porfolio"
                component={PortfolioScreen}
                options={{
                  tabBarIcon: ({ color }: TabBarIconProps) => (
                    <Icon name="portfolio" color={color} />
                  ),
                }}
              />
              <Tab.Screen
                name="Wallets"
                component={GenericScreen}
                options={{
                  tabBarIcon: ({ color }: TabBarIconProps) => (
                    <Icon name="wallet" color={color} />
                  ),
                }}
              />
              <Tab.Screen
                name="Market"
                component={GenericScreen}
                options={{
                  tabBarIcon: ({ color }: TabBarIconProps) => (
                    <Icon name="graph" color={color} />
                  ),
                }}
              />
              <Tab.Screen
                name="Alerts"
                component={GenericScreen}
                options={{
                  tabBarIcon: ({ color }: TabBarIconProps) => (
                    <Icon name="alert" color={color} />
                  ),
                }}
              />
              <Tab.Screen
                name="More"
                component={GenericScreen}
                options={{
                  tabBarIcon: ({ color }: TabBarIconProps) => (
                    <Icon name="more" color={color} />
                  ),
                }}
              />
            </Tab.Navigator>
          </NavigationContainer>
        </SafeAreaProvider>
      </SafeAreaView>
    </Provider>
  );
};

const styles = StyleSheet.create({
  appContainer: { flex: 1, backgroundColor: 'black' },
  scene: { backgroundColor: 'black' },
  screen: { flex: 1, alignItems: 'center', justifyContent: 'center' },
  text: { color: 'white', fontSize: 24 },
});

export default App;
