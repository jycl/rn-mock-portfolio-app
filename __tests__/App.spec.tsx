/**
 * @format
 */

import 'react-native';
import React from 'react';
import App, { GenericScreen } from '../App';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

describe('App', () => {
  it('renders correctly', () => {
    renderer.create(<App />);
  });

  it('GenericScreen with no route matches snapshot', () => {
    expect(renderer.create(<GenericScreen />)).toMatchSnapshot();
  });

  it('GenericScreen with route matches snapshot', () => {
    const props = { route: { name: 'Portfolio' } };
    expect(renderer.create(<GenericScreen {...props} />)).toMatchSnapshot();
  });
});
