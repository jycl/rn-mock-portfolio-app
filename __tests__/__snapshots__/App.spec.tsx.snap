// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App GenericScreen with no route matches snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 24,
      }
    }
  >
    Generic
  </Text>
</View>
`;

exports[`App GenericScreen with route matches snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 24,
      }
    }
  >
    Portfolio
  </Text>
</View>
`;
